cmd/metal-api/internal/service/switch-service.go

```diff
@@ -125,6 +136,46 @@ func (r switchResource) deleteSwitch(request *restful.Request, response *restful
 	}
 }
 
+func (r switchResource) notifySwitch(request *restful.Request, response *restful.Response) {
+	var requestPayload v1.SwitchNotifyRequest
+	err := request.ReadEntity(&requestPayload)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	id := request.PathParameter("id")
+	s, err := r.ds.FindSwitch(id)
+	if err != nil && !metal.IsNotFound(err) {
+		if checkError(request, response, utils.CurrentFuncName(), err) {
+			return
+		}
+	}
+
+	old := *s
+	sync := &metal.SwitchSync{
+		Time:     time.Now(),
+		Duration: requestPayload.Duration,
+	}
+
+	if requestPayload.Error == nil {
+		s.LastSync = sync
+	} else {
+		sync.Error = requestPayload.Error
+		s.LastSyncError = sync
+	}
+
+	err = r.ds.UpdateSwitch(&old, s)
```

Comment:
This might fail because we have a quite high update frequency on the switch entity, but i think this is ok.